app_dart/lib/src/request_handlers/scheduler/batch_backfiller.dart

```diff
@@ -99,15 +102,50 @@ class BatchBackfiller extends RequestHandler {
       });
       // Schedule all builds asynchronously.
       // Schedule after db updates to avoid duplicate scheduling when db update fails.
-      await Future.wait<void>(backfillRequestList(backfill));
+      await _scheduleWithRetries(backfill);
     } catch (error) {
       log.severe('Failed to update tasks when backfilling: $error');
     }
   }
 
+  /// Schedules tasks with retry when hitting pub/sub server errors.
+  Future<void> _scheduleWithRetries(List<Tuple<Target, FullTask, int>> backfill) async {
+    const RetryOptions retryOptions = Config.schedulerRetry;
+    try {
+      await retryOptions.retry(
+        () async {
+          final List<List<Tuple<Target, Task, int>>> tupleLists =
+              await Future.wait<List<Tuple<Target, Task, int>>>(backfillRequestList(backfill));
+          if (tupleLists.any((List<Tuple<Target, Task, int>> tupleList) => tupleList.isNotEmpty)) {
+            backfill = _updateBackfill(backfill, tupleLists);
+            throw InternalServerError('Failed to backfill ${backfill.length} targets.');
+          }
+        },
+        retryIf: (Exception e) => e is InternalServerError,
+      );
+    } catch (e) {
+      log.severe('Failed to backfill ${backfill.length} targets.');
+    }
+  }
+
+  /// Updates the [backfill] list with those that fail to get scheduled.
+  ///
+  /// [tupleLists] maintains the same tuple order as those in [backfill].
+  /// Each element from [backfill] is encapsulated as a list in [tupleLists] to prepare for
+  /// [scheduler.luciBuildService.schedulePostsubmitBuilds].
+  _updateBackfill(List<Tuple<Target, FullTask, int>> backfill, List<List<Tuple<Target, Task, int>>> tupleLists) {
```

Comment:
I did not know you could omit the return type in dart like this.

Reply:
Good catch. This is not a good style. Added the return type.